extmod/network_cyw43.c

```diff
@@ -321,6 +321,13 @@ STATIC mp_obj_t network_cyw43_ifconfig(size_t n_args, const mp_obj_t *args) {
 }
 MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(network_cyw43_ifconfig_obj, 1, 2, network_cyw43_ifconfig);
 
+mp_obj_t network_nic_config(struct netif *netif, size_t n_args, const mp_obj_t *args, mp_map_t *kwargs);
```

Comment:
> So you would call even the method that is used to implement nic.ipconfig "static"?

No, this cannot be `static` because it's in a different compilation unit. I agree the convention is a bit loose (but I think it's at least consistent across all of the network code) is that the `mod_` prefix is used only on non-static methods.

So the convention as it stands is:

* Generally, in modfoo.c, the methods should be named `foo_bar`, and the wrapper objects named `foo_bar_obj`.
* Where possible, both the methods and the wrappers should be `static`.
* Where these functions need to call into hooks provided (e.g. by the port or elsewhere), the name should be something like `mp_foo_`.

There is remnants of an older convention (modjson.c, modbinascii.c, modbtree.c) where `mod_` was used as a prefix for methods. In general, almost all of these are `static`, because typically most modules are self-contained.

However, the two conventions seem to have blended in the network code, and the current situation is that the `mod_network_` prefix is used on non-static methods (i.e. because they are required to be implemented elsewhere), and just `network_` on static methods (the default).

Anyway, in this case please just match the existing way that `mod_network_nic_ifconfig` works (non-static, `mod_` prefix).

> Concluding, I will name the static method mod_network_ipconfig and the non-static method network_nic_ipconfig

Both of these methods are non-static, and they both work in the same way.

If I were to overhaul this convention, the main thing I would fix here is that both of these methods (and the existing `mod_network_nic_ifconfig` should have `lwip` somewhere in their name. In other works, `network_lwip.c` is a collection of helper methods that can be used by lwip-based nics. (And the especially confusing thing here is that `network_lwip.c` is not like, say, `network_cyw43.c`, rather it provides helper methods for e.g. `network_cyw43.c`)

> These are both static methods, one is prefixed with mod this other isn't. (gathering from the TODO, in L125), this probably occurred during refactoring.

`network_country` _is_ `static`, but `mod_network_hostname` is non-static. Nothing else calls `network_country` other than via Python, whereas `mod_network_hostname` is used in a bunch of places around the codebase.

Perhaps the confusion here (and what that TODO on L125) is about, is that the _wrapper_ object (the actual Python function object) is non-static. This allows a locals dict elsewhere to use the wrapper directly.

Also possibly we're talking at cross-purposes about the meaning of `static`. I am exclusively using it in the sense of the C keyword (i.e. private to this compilation unit). The fact that `network.ipconfig` is "static" in sort of the Python/C++ sense (versus an instance method like nic.ipconfig) has no bearing on the "mod_" prefix.